### PR TITLE
Update p2553

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

File: P2553/P2553.md [hunk 1]
```diff
@@ -199,17 +199,24 @@ as of [P0009](https://wg21.link/P0009) R15.
...
-```c++
-template<size_t... Extents>
-  class extents;
-```
+  ```c++
+  template<size_t... Extents>
+    class extents;
+
+  template<size_t Rank>
+    using dextents = see below;
```
Comment (dalg24) [inline P2553/P2553.md:207]: Did you mean to indent with two spaces?

Comment (crtrott) [reply P2553/P2553.md:207]: yeah

File: P2553/P2553.md [hunk 2]
```diff
@@ -220,11 +227,16 @@ template<class SizeT, size_t... Extents>
 class extents {
 public:
   using size_type = SizeT;
+  using rank_type = size_t;
```
Comment (dalg24) [inline P2553/P2553.md:230]: Do we need to spell that addition out in the description?

File: P2553/P2553.md [hunk 3]
```diff
@@ -438,14 +429,44 @@ with:
...
+
+* After paragraph 5 add new paragraph:
+
+*Preconditions*: `other.required_span_size()` is a representable value of `size_type` ([basic.fundamental]).
+
+* Change paragraph 8 to:
+
+*Preconditions*: 
+
+  - If `extents_type::rank() > 0` is `true` then for all $r$ in the range $[$`0, extents_type::rank()`$)$, `other.stride(r)` equals `extents().`_fwd-prod-of-extents_`(r)`, and
```
Comment (mhoemmen) [inline P2553/P2553.md:448]: Would you consider not putting the comma in the mathematical interval in code font?
```suggestion
  - If `extents_type::rank() > 0` is `true` then for all $r$ in the range $[$`0`, `extents_type::rank()`$)$, `other.stride(r)` equals `extents().`_fwd-prod-of-extents_`(r)`, and
```

File: P2553/P2553.md [hunk 4]
```diff
@@ -438,14 +429,44 @@ with:
...
+* After paragraph 5 add new paragraph:
 
-* In paragraph 3 add a new sub bullet
+*Preconditions*: `other.required_span_size()` is a representable value of `size_type` ([basic.fundamental]).
+
+* Change paragraph 8 to:
+
+*Preconditions*: 
+
+  - If `extents_type::rank() > 0` is `true` then for all $r$ in the range $[$`0, extents_type::rank()`$)$, `other.stride(r)` equals `extents().`_rev-prod-of-extents_`(r+1)`, and
```
Comment (mhoemmen) [inline P2553/P2553.md:466]: Please see above.
```suggestion
  - If `extents_type::rank() > 0` is `true` then for all $r$ in the range $[$`0`, `extents_type::rank()`$)$, `other.stride(r)` equals `extents().`_rev-prod-of-extents_`(r+1)`, and
```

**Review by mhoemmen — COMMENTED** [review]
Just a couple comma-in-code-font questions : - )

**Pull request merged**